hxsl/Ast.hx

```diff
@@ -313,6 +314,24 @@ enum TGlobal {
 	UnpackUnorm4x8;
 }
 
+enum SyntaxArgAccess {
+	Read;
+	Write;
+	ReadWrite;
+}
+
+typedef SyntaxArg = {
+	e: TExpr,
+	access: SyntaxArgAccess,
+}
+
+enum SyntaxTarget {
+	/** Applies to any target language **/
+	Code;
+	Glsl;
+	Hlsl;
+}

```

Comment:
I think SyntaxTarget should be a String, so adding more targets doesn't require to change HxSL AST

Reply:
Done